Examples: init store in main, access through provider class (#316)
objectbox/objectbox-dart: objectbox/example/README.md

### Flutter apps

For example, open the `Store` in a small helper class like this:

```dart
import 'objectbox.g.dart'; // created by `flutter pub run build_runner build`

class ObjectBox {
  Store? _store;

  /// Initialize the store.
  Future<void> init() async {
    // Future<Store> openStore() {...} is defined in the generated objectbox.g.dart
    final store = await openStore();
    _store = store;
  }

  /// Returns the open Store for this app or throws.
  Store get store {
    final store = _store;
    if (store != null) {
      return store;
    } else {
      throw Exception('Store was not initialized on app launch');
    }
  }
}
```

The best time to initialize ObjectBox is when your app starts. 
We suggest to do it in your app's `main()` function:

```dart

> /// Provides access to the ObjectBox Store throughout the app.
> late ObjectBox objectbox;
> 
> Future<void> main() async {
>   // This is required so ObjectBox can get the application directory
>   // to store the database in.
>   WidgetsFlutterBinding.ensureInitialized();
> 
>   objectbox = ObjectBox();
>   await objectbox.init();
> 
>   runApp(MyApp());
> }

Reviewer: What about the following, without the wrapper?
```suggestion
/// Provides access to the ObjectBox Store throughout the app.
late final Store objectbox;

Future<void> main() async {
  objectbox = await openStore();
  runApp(MyApp());
}
```

Also I'd suggest moving `WidgetsFlutterBinding.ensureInitialized()` call to the generated `openStore()`. No need to bother users with that and it should be pretty cheap.

Author: Thanks! That was my other suggestion. But decided to use a wrapper class that initializes some re-usable things like boxes and queries. Also to possibly have a single class if a demo test gets added in the future.

As for moving `WidgetsFlutterBinding.ensureInitialized()` into `openStore()`: would this then still work with from a test? But I guess can only call it if `directory` is null.
